aspnet/security/authentication/cookieauthentication.rst

```diff
@@ -0,0 +1,166 @@
+     {
+         // Pull database from registered DI services.
+         var userRepository = context.HttpContext.RequestServices.GetRequiredService<IUserRepository>();
+         var userPrincipal = context.Principal;
+
+         // Look for the last changed claim.
+         string lastChanged;
+         lastChanged = (from c in userPrincipal.Claims
+                        where c.Type == "LastUpdated"
+                        select c.Value).FirstOrDefault();
+
+         if (string.IsNullOrEmpty(lastChanged) ||
+             !userRepository.ValidateLastChanged(userPrincipal, lastChanged))
+         {
+             context.RejectPrincipal();
+             await context.HttpContext.Authentication.SignOutAsync("MyCookieMiddlewareInstance");
+         }
+     }
+ }
+
+This would then be wired up during cookie middleware configuration
+
+.. code-block:: c#
+
+ app.UseCookieAuthentication(options =>
+ {
+     options.Events = new CookieAuthenticationEvents
+     {
+         // Set other options
+         OnValidatePrincipal = LastChangedValidator.ValidateAsync
+     };
+ });
+
+If you want to non-destructively update the user principal, for example, their name might have been updated, a decision which doesn't affect security in any way you can call ``context.ReplacePrincipal()`` and set the ``context.ShouldRenew`` flag to ``true``.
+
+.. _security-authentication-cookie-options:
+
+Controlling cookie options
+--------------------------
+
+The ``CookieAuthenticationOptions`` class comes with various configuration options to enable you to fine tune the cookies created.
+
+* **ClaimsIssuer** - the issuer to be used for the `Issuer <https://msdn.microsoft.com/en-us/library/microsoft.identitymodel.claims.claim.issuer.aspx>`_ property on any claims created by the middleware.
+* **CookieDomain** - the domain name the cookie will be served to. By default this is the host name the request was sent to. The browser will only serve the cookie to a matching host name. You may wish to adjust this to have cookies available to any host in your domain. For example setting the cookie domain to ``.contoso.com`` will make it available to ``contoso.com``, ``www.contoso.com``, ``staging.www.contoso.com`` etc.
+* **CookieHttpOnly** - a flag indicating if the cookie should only be accessible to servers. This defaults to ``true``. Changing this value may open your application to cookie theft should your application have a Cross Site Scripting bug.
+* **CookiePath** - this can be used to isolate applications running on the same host name. By default any cookies generated will have a path of ``\`` which makes them available to any request path on your website. If you have an app running in ``/app1`` and want to limit the cookies issued to just be sent to that application then you should set the ``CookiePath`` property to ``/app1``. The cookie will now only be available to requests to ``//app1`` or anything underneath it.
```

Comment:
`any cookies generated will have a path of ``\`` which makes them available to any request path on your website.`... that's not totally true, actually: when no `CookiePath` is explicitly set, the path base of the current request is used as a fallback, which means that if your cookie middleware instance is inside a `Map` registration,  the path associated with `Map` is reused for the `CookiePath`. Same thing for virtual directories when running behind IIS (well, currently, it's broken because of the new hosting model... :smile:)